src/docs/sdk/performance/opentelemetry.mdx

```diff
@@ -18,7 +18,220 @@ When Sentry performance monitoring was initially introduced, OpenTelemetry was i
+    this._options = options;
+  }
+
+  onStart(otelSpan: OpenTelemetrySpan): void {
+    const hub = Sentry.getCurrentHub();
+    const scope = hub.getScope();
+
+    if (!hub || !scope) return;
+
+    const otelSpanId = span.spanContext().spanId;
+
+    const parentSpan = scope.getSpan();
+    if (parentSpan) {
+      // If the parentSpan exists, we create a new span as a child of the parentSpan
+      const childSpan = parentSpan.startChild({
+        description: otelSpan.name,
+        instrumenter: 'otel',
+        // don't define op because we will do that on span.finish
+      });
+
+      if (childSpan) {
+          // We store the otelSpan and it's parent in the map
+          this._map[otelSpanId] = [childSpan, parentSpan];
+          scope.setSpan(childSpan);
+      }
+
+    } else {
+      // Will grab Dynamic Sampling Context and sentry-trace header if available
+      const traceCtx = getTraceData(otelSpan);
+      // If the parentSpan doesn't exist, we create a new transaction
+      const transaction = Sentry.startTransaction({
+        name: otelSpan.name,
+        ...traceCtx,
+        instrumenter: 'otel'
+        // don't define op because we will do that on span.finish
+      });
+
+      if (transaction) {
+          // We store the otelSpan and it's parent in the map
+          this._map[otelSpanId] = [transaction, undefined];
+          scope.setSpan(transaction);
+      }
+    }
+  }
+
+  onEnd(otelSpan: OpenTelemetrySpan): void {
+    const hub = Sentry.getCurrentHub();
+    const scope = hub.getScope();
+
+    if (!hub || !scope) return;
+
+    const otelSpanId = otelSpan.spanContext().spanId;
+    const mapVal = this._map[otelSpanId];
+
+    if (mapVal) {
+      const [sentrySpan, parentSpan] = mapVal;
+
+      // Update op, description, and attach otel resource attributes
+      updateSpanWithOtelData(sentrySpan, otelSpan);
+
+      // We finish the span
+      sentrySpan.finish();
+      // We set the parent span as the active span if parent span is defined
+      if (parentSpan) {
+          scope.setSpan(parentSpan);
+      }
+      // We remove the span from the map
+      delete this._map[otelSpanId];
+    }
+  }
+}
+```
+
+Users are required to add this `SentrySpanProcessor` to their OpenTelemetry SDK initialization logic to make this work, like so. Individual SDK implementations might be a little different.
+
+```ts
+import { NodeSDK } from "@opentelemetry/sdk-node";
+import { Resource } from "@opentelemetry/resources";
+import * as Sentry from "@sentry/node";
+import { SentrySpanProcessor } from "@sentry/opentelemetry-node";
+
+Sentry.init({
+  /// ...
+});
+
+const sdk = new NodeSDK({
+  resource: new Resource({
+    "service.name": "my-service",
+    "service.version": "1.0.0",
+  }),
+  spanProcessor: new SentrySpanProcessor(),
+});
+```
+
+### Step 2: Define `getTraceData`
+
+We want to make sure a transaction is started with information from the `sentry-trace` and `baggage` headers.
+
+TODO @neel
```

Comment:
In an internal call Daniel K. just mentioned `SpanPropagator` which could be used for this. But I guess you're already aware.

Reply:
Yup, we'll chat more about this next week.